packages/flutter_tools/lib/src/ios/migrations/uiapplicationmain_deprecation_migration.dart

```diff
@@ -0,0 +1,49 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../../base/file_system.dart';
+import '../../base/project_migrator.dart';
+import '../../xcode_project.dart';
+
+const String _appDelegateFileBefore = r'''
+@UIApplicationMain
+@objc class AppDelegate''';
+
+const String _appDelegateFileAfter = r'''
+@main
+@objc class AppDelegate''';
+
+/// Replace the deprecated `@UIApplicationMain` attribute with `@main`.
+///
+/// See:
+/// https://github.com/apple/swift-evolution/blob/main/proposals/0383-deprecate-uiapplicationmain-and-nsapplicationmain.md
+class UIApplicationMainDeprecationMigration extends ProjectMigrator {
+  UIApplicationMainDeprecationMigration(
+    IosProject project,
+    super.logger,
+  ) : _appDelegateSwift = project.appDelegateSwift;
+
+  final File _appDelegateSwift;
+
+  @override
+  Future<bool> migrate() async {
```

Comment:
```suggestion
  Future<void> migrate() async {
```
I'm not sure why https://github.com/flutter/flutter/blob/8c12ba84705ba4a8056f83928fb83b251f194101/packages/flutter_tools/lib/src/ios/migrations/remove_bitcode_migration.dart#L19 has `bool`...

Reply:
Excellent catch! I fixed this and updated `RemoveBitcodeMigration` as well

Reply:
Tests using `RemoveBitcodeMigration` need to updated to no longer expect bool

Example:
https://github.com/flutter/flutter/blob/839a6b3c8747557dd858b0237c482219573e2cea/packages/flutter_tools/test/general.shard/ios/ios_project_migration_test.dart#L906-L910

Which should fix the failing tests:
```
error • This expression has a type of 'void' so its value can't be used • packages/flutter_tools/test/general.shard/ios/ios_project_migration_test.dart:910:16 • use_of_void_result
   error • This expression has a type of 'void' so its value can't be used • packages/flutter_tools/test/general.shard/ios/ios_project_migration_test.dart:926:16 • use_of_void_result
   error • This expression has a type of 'void' so its value can't be used • packages/flutter_tools/test/general.shard/ios/ios_project_migration_test.dart:947:16 • use_of_void_result
   error • This expression has a type of 'void' so its value can't be used • packages/flutter_tools/test/general.shard/ios/ios_project_migration_test.dart:1432:14 • use_of_void_result
   error • This expression has a type of 'void' so its value can't be used • packages/flutter_tools/test/general.shard/ios/ios_project_migration_test.dart:1461:14 • use_of_void_result
   error • This expression has a type of 'void' so its value can't be used • packages/flutter_tools/test/general.shard/ios/ios_project_migration_test.dart:1490:14 • use_of_void_result
```

```
Failed to load "test/general.shard/ios/ios_project_migration_test.dart":
   test/general.shard/ios/ios_project_migration_test.dart:910:16: Error: This expression has type 'void' and can't be used.
           expect(await migration.migrate(), isTrue);
                  ^
   test/general.shard/ios/ios_project_migration_test.dart:926:16: Error: This expression has type 'void' and can't be used.
           expect(await migration.migrate(), isTrue);
                  ^
   test/general.shard/ios/ios_project_migration_test.dart:947:16: Error: This expression has type 'void' and can't be used.
           expect(await migration.migrate(), isTrue);
                  ^
   test/general.shard/ios/ios_project_migration_test.dart:1432:14: Error: This expression has type 'void' and can't be used.
         expect(await migration.migrate(), isTrue);
                ^
   test/general.shard/ios/ios_project_migration_test.dart:1461:14: Error: This expression has type 'void' and can't be used.
         expect(await migration.migrate(), isTrue);
                ^
   test/general.shard/ios/ios_project_migration_test.dart:1490:14: Error: This expression has type 'void' and can't be used.
         expect(await migration.migrate(), isTrue);
```

Reply:
Fixed! (... and ran the the tests locally this time 😅)